### DISCUSSION TITLE
C++ source files should include only C++ wrappers over C headers instead of plain C headers

### DISCUSSION THREAD

**Discussion by cflaviu** [managarm/managarm#327] [General]
Examples:
-  \<cstdint\> instead of \<stdint.h\>
-  \<cassert\> instead of \<assert.h\>

**Comment by avdgrinten** [comment]
I think this is not necessary in the general case (it doesn't add anything but adds a bit of code churn). There are several cases where C++ headers are needed because they add functionality though: for example, `<cstddef>` adds `std::byte` and `nullptr_t`.

We control the C headers anyway, so there is little reason to replace them by C++ headers.

**Reply by cflaviu** [reply]
Well I guess in the future those C++ headers will be needed anyway. It's more a matter of time from my point of view.

**Reply by avdgrinten** [reply]
Sure, when we need functionality, we will switch. But replacing all C style headers now doesn't have any advantage.
